quickwit-cli/src/lib.rs

```diff
@@ -84,61 +84,81 @@ impl PartialEq for CreateIndexArgs {
 
 impl CreateIndexArgs {
     pub fn new(
+        metastore_uri: String,
         index_uri: String,
         index_config_path: PathBuf,
-        metastore_uri: String,
         overwrite: bool,
     ) -> anyhow::Result<Self> {
-        let json_file = std::fs::File::open(index_config_path)?;
+        let json_file = std::fs::File::open(index_config_path.clone()).map_err(|error| {
```

Comment:
`map_err(|err| anyhow::anyhow!(...))?` can be replaced with the `anyhow` helper `with_context(...)?`